frontend/src/components/elements/Metric/Metric.test.tsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react"
+import { Metric as MetricProto } from "src/autogen/proto"
+import { mount } from "src/lib/test_util"
+import Metric, { MetricProps } from "./Metric"
+
+const getProps = (elementProps: Partial<MetricProto> = {}): MetricProps => ({
+  element: MetricProto.create({
+    color: MetricProto.MetricColor.RED,
+    direction: MetricProto.MetricDirection.UP,
+    ...elementProps,
+  }),
+})
+
+describe("Metric element", () => {
+  it("renders metric as expected", () => {
+    const props = getProps()
+    const wrapper = mount(<Metric {...props} />)
+    expect(wrapper).toBeDefined()
+  })
+
+  it("renders direction icon based on props", () => {
+    const props = getProps()
+    const wrapper = mount(<Metric {...props} />)
+    expect(wrapper.find("StyledMetricDeltaText").find("svg")).toBeDefined()
+  })
+
+  it("renders direction icon based on props", () => {
+    const props = getProps({
+      color: MetricProto.MetricColor.GREEN,
+      direction: MetricProto.MetricDirection.DOWN,
+    })
+    const wrapper = mount(<Metric {...props} />)
+    expect(wrapper.find("StyledMetricDeltaText").find("svg")).toBeDefined()
+  })
```

Comment:
nit: You'll also want to verify that the right arrow is displayed in both of these cases (and also may want to make the test names slightly more descriptive to include which direction is being tested)